### ISSUE TITLE
Exception: [Errno 32] Broken pipe

### ISSUE THREAD

**Issue by vengin** [SublimeLinter/SublimeLinter#1518]
All of a sudden started receiving the `Exception: [Errno 32] Broken pipe`.
OS:  Win7 Enterprise (x64), Version 6.1 (Build 7601)
SublimeText version: 3.1.1, Build 3176

I am using SublimeLinter with [SublimeLinter-contrib-vcom](https://github.com/jevogel/SublimeLinter-contrib-modelsim/) plugin and it was working fine. All of a sudden I've started getting the following error:
```
SublimeLinter: sublime_linter.py:252  Delay linting 'dpram_fifo.vhd' for 0.0s
SublimeLinter: linter.py:889          vcom: checking lint mode 'save' vs lint reason 'on_save'
SublimeLinter: #4 linter.py:906       vcom: linting 'dpram_fifo.vhd'
SublimeLinter: #4 linter.py:1263      Running ...

  D:\Proj\syna\Dev\seg_sublime_prj/../seg_msim_prj/msim_lib  (working dir)
  > type ..\..\src_x\dpram_fifo.vhd | C:\PF\modeltech64_10.5\win64\vcom.exe -check_synthesis -2008 -suppress 1074,1236 D:\Proj\tmp\Dev\src_x\dpram_fifo.vhd

SublimeLinter: #4 linter.py:1277      WARNING: Exception: [Errno 32] Broken pipe
SublimeLinter: #4 linter.py:949       vcom: no output
SublimeLinter: sublime_linter.py:449  Linting 'dpram_fifo.vhd' took 0.17s
```
I.e. I'm trying to lint VHDL file `drpram_fifo.vhd` with ModelSim `vcom` compiler. If I do it manually, from command prompt, everything is fine:
```> C:\PF\modeltech64_10.5\win64\vcom.exe -check_synthesis -2008 -suppress 1074,1236 D:\Proj\syna\Dev\src_x\dpram_fifo.vhd
Model Technology ModelSim SE-64 vcom 10.5 Compiler 2016.02 Feb 13 2016
Start time: 08:29:57 on Nov 08,2018
vcom -check_synthesis -2008 -suppress 1074,1236 D:\Proj\tmp\Dev\src_x\dpram_fifo.vhd
-- Loading package STANDARD
-- Loading package TEXTIO
-- Loading package std_logic_1164
-- Loading package std_logic_arith
-- Loading package STD_LOGIC_UNSIGNED
-- Loading package MATH_REAL
-- Compiling entity dpram_fifo
-- Compiling architecture beh of dpram_fifo
-- Loading entity simple_dpram_single_clock
End time: 08:29:57 on Nov 08,2018, Elapsed time: 0:00:00
Errors: 0, Warnings: 0
```

And the same vhd file was working before, and other files (so far) do not produce this exception.

What might be causing it?

_Originally posted by @vengin in https://github.com/SublimeLinter/SublimeLinter/issues/1327#issuecomment-436882082_

**Comment by vengin** [comment]
Just wanted to add, that I've started noticing the same `Exception: [Errno 32] Broken pipe` with other files as well.

**Comment by kaste** [comment]
Boy, I'm starring at this issue way too long bc from reading your log it is easy to see why that fails. But the plugin code https://github.com/jevogel/SublimeLinter-contrib-modelsim/blob/master/linter.py didn't match the behavior. So I thought it is our fault and a bug. But it's just that they forget to cut a release.

TLDR;  The `master` of the plugin should work. Please trigger the repo owner to make a new release. (Or install it using `git clone`)

**Comment by kaste** [comment]
Explainer: 

The old version of the plugin uses STDIN to feed the current code to `vcom` but the `vcom` linter doesn't _read_ from STDIN. STDIN is buffered. If the buffer is full bc nobody reads from it, the BrokenPipe gets thrown. 

For small files, it doesn't fail bc the the file fits in the buffer.

**Comment by vengin** [comment]
> The `master` of the plugin should work. Please trigger the repo owner to make a new release. (Or install it using `git clone`)

Do you mean SublimeLInter plugin?

**Comment by vengin** [comment]
Just tried the SublimeLInter `master` and the issue remains.

**Comment by braver** [comment]
> Do you mean SublimeLInter plugin?

No, modelsim

**Comment by vengin** [comment]
Oh, I see. After replacing my local version of the `SublimeLinter-contrib-vcom` plugin with the `master` the problem went away. It turns out there is a [commit, which actually describes and solves the problem](https://github.com/jevogel/SublimeLinter-contrib-modelsim/commit/b3ab7152f7f22921d99a2adc671bd03b4a8d3e4c). Howerver, I wasn't aware of it, since there was no release, and my local plugin version lacked the `tempfile_suffix` fix.
Thanks.

**Comment by kaste** [comment]
🎉

**Comment by jevogel** [comment]
v1.2.0 of `SublimeLinter-contrib-modelsim` (previously known as `SublimeLinter-contrib-vcom`) is now released with the `tempfile_suffix` included. Thanks for reporting this issue and I apologize for the delay in getting this released.
